.github/workflows/reusable-on-code-change.yml

```diff
@@ -0,0 +1,198 @@
+name: Reusable Workflow (On Code Change)
+on:
+  workflow_call:
+    inputs:
+        static_analysis_instance_type:
+            type: string
+            default: "ubuntu-latest"
+        build_instance_type:
+            type: string
+            default: "ubuntu-latest"
+        sonar_analysis_instance_type:
+            type: string
+            default: "ubuntu-latest"
+        integration_tests_instance_type:
+            type: string
+            default: "self-hosted"
+        tests_standalone:
+            type: boolean
+            default: false
+        tests_cluster:
+            type: boolean
+            default: false
+        node_version:
```

Comment:
Same comment here (and there will likely be more below), if a parameter apply only to one job, we should highlight this in some way (i.e. we're not setting the node_version of the integration tests)